src/CalendarDate.ts

```diff
@@ -0,0 +1,171 @@
+import { CalendarDateRange } from './CalendarDateRange';
+
+const DAY_IN_SECONDS = 86400;
+const daysInMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+
+export class CalendarDate {
+  readonly year!: number;
+
+  /**
+   * Month of the year starting from 0
+   */
+  readonly month!: number;
+
+  /**
+   * Day of the month starting from 1
+   */
+  readonly day!: number;
+
+  /**
+   * Seconds passed since the unix epoch. Always calculated with a time and timezone set to T00:00:00.00Z.
+   */
+  readonly unixTimestampInSeconds!: number;
+
+  constructor(isoString: string);
+  constructor(unixTimestampSeconds: number);
+  constructor(year: number, month: number, day: number);
+  /**
+   * If you supply a Date object, it will take the values for year month and day for the local time.
+   */
+  constructor(input1: string | number, input2?: number, input3?: number) {
+    let year = 0;
+    let month = 0;
+    let day = 0;
+    if (typeof input1 === 'string') {
+      return CalendarDate.parseString(input1);
+    }
+    if (typeof input2 === 'number' && typeof input3 === 'number') {
+      year = input1;
+      month = input2;
+      day = input3;
+      if (year < 0 || year > 9999) {
+        throw new Error(
+          `CalendarDate Validation Error: Input year ${year} is not valid. Year must be a number between 0 and 9999.`,
+        );
+      }
+      if (month < 0 || month > 11) {
+        throw new Error(
+          `CalendarDate Validation Error: Input month ${month} is not valid. Month must be a number between 0 and 11.`,
+        );
+      }
+      if (day < 1 || day > 31) {
```

Comment:
you are catching in line 60 that day can be bigger then max days of month. Why we need a guarding block here? The same we can also do for month number